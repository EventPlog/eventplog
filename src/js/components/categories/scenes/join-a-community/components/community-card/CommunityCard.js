import React from 'react'
import { Card, Icon } from 'semantic-ui-react'
import Button from 'js/components/shared/button'

const CommunityCard = ({
  name,
  description,
  no_of_followers,
  no_of_upcoming_events,
  community_focus,
}) => (
  <div>
    <div className="header">
      { name }
    </div>
    <div className="description">
      { description }
    </div>
    <div className="meta">
      <div className="item">
        { no_of_followers }
      </div>
      <div className="item">
        { no_of_upcoming_events }
      </div>
      <div className="item">
        { community_focus }
      </div>
      <Button>Follow</Button>
    </div>
  </div>
)

export default CommunityCard