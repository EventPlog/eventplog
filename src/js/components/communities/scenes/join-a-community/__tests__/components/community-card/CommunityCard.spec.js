import React from 'react'
import CommunityCard from '../../../components/community-card'
import { shallow, mount} from 'enzyme'

describe('CommunityCard', () => {
  test('it renders correctly', () => {
    const wrapper = shallow( <CommunityCard /> )
    expect(wrapper).toMatchSnapshot()
  })

  test('it renders based on props', () => {
    // setup
    const props = {
      name: 'Dev C Lagos',
      description: 'An amazing community',
      no_of_followers: 20,
      no_of_upcoming_events: 4,
      community_focus: 'technology'
    }

    // action
    const wrapper = mount( <CommunityCard {...props} />)

    // assert
    expect(wrapper.find('.header').text()).toEqual(props.name)
    expect(wrapper.find('.description').text()).toEqual(props.description)
  })
})