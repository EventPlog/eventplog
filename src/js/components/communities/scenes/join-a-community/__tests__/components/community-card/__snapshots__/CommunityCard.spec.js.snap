// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunityCard it renders correctly 1`] = `
<div>
  <div
    className="header"
  />
  <div
    className="description"
  />
  <div
    className="meta"
  >
    <div
      className="item"
    />
    <div
      className="item"
    />
    <div
      className="item"
    />
    <Button>
      Follow
    </Button>
  </div>
</div>
`;
